tests/supervisors/test_multiprocess.py

```diff
@@ -169,3 +174,71 @@ def test_multiprocess_sigttou() -> None:
     assert len(supervisor.processes) == 1
     supervisor.signal_queue.append(signal.SIGINT)
     supervisor.join_all()
+
+
+T = TypeVar("T")
+
+
+@dataclasses.dataclass
+class Box(Generic[T]):
+    v: T
+
+
+async def lb_app(
+    d: multiprocessing.managers.DictProxy,
+    started: threading.Event,
+    scope: Scope,
+    receive: ASGIReceiveCallable,
+    send: ASGISendCallable,
+) -> None:  # pragma: py-darwin pragma: py-win32
+    if scope["type"] == "lifespan":
+        await receive()
+        scope["state"]["count"] = box = Box(0)
+        await send({"type": "lifespan.startup.complete"})
+        started.set()
+        await receive()
+        d[os.getpid()] = box.v
+        await send({"type": "lifespan.shutdown.complete"})
+        return
+
+    scope["state"]["count"].v += 1
+    headers = [(b"content-type", b"text/plain")]
+    await send({"type": "http.response.start", "status": 200, "headers": headers})
+    await send({"type": "http.response.body", "body": b"hello"})
+
+
+@pytest.mark.skipif(
+    not ((sys.platform == "linux" and hasattr(socket, "SO_REUSEPORT")) or hasattr(socket, "SO_REUSEPORT_LB")),
+    reason="unsupported",
+)
+def test_multiprocess_socket_balance() -> None:  # pragma: py-darwin pragma: py-win32
+    with multiprocessing.Manager() as m:
+        started = m.Event()
+        d = m.dict()
+        app = functools.partial(lb_app, d, started)
+        config = Config(app=app, workers=2, socket_load_balance=True, port=0, interface="asgi3")
```

Comment:
this needs a test for two independent calls to Multiprocess(...).run() running on the same port:

```python
def test_bind_to_used_port():
    port = ephemeral_port_reserve.reserve() 
    config = Config(app=app, workers=2, socket_load_balance=True, port=port, interface="asgi3")
    with multiprocessing.get_context("spawn").Pool(max_workers=1) as pool:
        try:
            f1 = pool.apply_async(uvicorn.main, ...)
            # wait for started
            with pytest.raises(OSError, match="already bound"):
                uvicorn.main(...)
        finally: pool.terminate()
```